### PR TITLE
CONN-1912 QA fix and cert expiry color legends

### PR DESCRIPTION
**Pull request by pvenkatakrishnan1**

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 1]
```diff
@@ -266,6 +272,58 @@ private void importSelectedCertificate() {
         }
     }
 
+    public void updateSelectedCertificate() {
+        LOG.info("updateSelectedCertificate");
+        if (selectedTSCertificate != null){
+            String alias = selectedTSCertificate.getAlias();
+            if (StringUtils.isNotBlank(alias) && !service.isAliasInUse(alias, service.fetchTrustStores())) {
+                try {
+                    boolean updateStatus = service.updateCertificateTS(oldAlias, selectedTSCertificate);
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: Use inline method call like this: postUpdateAction(service.updateCertificateTS(oldAlias, selectedTSCertificate), VIEW_CERT_ERR_MSG);

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 2]
```diff
@@ -266,6 +272,58 @@ private void importSelectedCertificate() {
...
+            } else {
+                selectedTSCertificate.setAlias(oldAlias);
+                HelperUtil.addMessageError(VIEW_CERT_ERR_MSG, "Alias already in use");
+            }
+        }
+    }
+
+    public void updateSelectedCertificateKS() {
+        LOG.info("updateSelectedCertificate");
+        if (selectedTSCertificate != null) {
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: move the common logic in a private method and call the method from both updateSelectedCertificateKS and updateSelectedCertificate

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 3]
```diff
@@ -266,6 +272,58 @@ private void importSelectedCertificate() {
...
+            }
+        }
+    }
+
+    /**
+     * @param updateStatus
+     */
+    private void postUpdateAction(boolean updateStatus, String uiElement) {
+        if (updateStatus) {
+            truststores = service.refreshTrustStores();
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: both truststore and keystore are getting refreshed here. You only need to refresh one.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 4]
```diff
@@ -284,4 +342,24 @@ private void fetchKeyStore() {
     private void fetchTrustStore() {
         truststores = service.fetchTrustStores();
     }
+
+    public Certificate viewCertificate() {
+        if (selectedTSCertificate != null) {
+            oldAlias = selectedTSCertificate.getAlias();
+            RequestContext.getCurrentInstance().execute("PF('viewCertDlg').show();");
+        } else {
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: viewCertificate and viewCertificateKS are basically the same method. You can move the common functionality into a private method and pass the dialog string as an argument to that method.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java [hunk 5]
```diff
@@ -269,4 +271,87 @@ public boolean deleteCertificateFromTrustStore(String alias) throws
...
+        final String storeLoc = trustStoreProperties.get(TRUST_STORE_KEY);
+        final String passkey = trustStoreProperties.get(TRUST_STORE_PASSWORD_KEY);
+        FileInputStream is = null;
+        FileOutputStream os = null;
+        boolean isUpdateSuccessful = false;
+        try {
+            KeyStore tstore = KeyStore.getInstance(storeType);
+            if (!PKCS11_TYPE.equalsIgnoreCase(storeType)) {
+                is = new FileInputStream(storeLoc);
+            }
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java]: Factor out the common logic in a private method for both updateCertificateTS and updateCertificateKS

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java [hunk 6]
```diff
@@ -269,4 +271,87 @@ public boolean deleteCertificateFromTrustStore(String alias) throws
...
+        tstore.deleteEntry(oldAlias);
+        os = new FileOutputStream(storeLoc);
+        tstore.setCertificateEntry(cert.getAlias(), cert.getX509Cert());
+        tstore.store(os, passkey.toCharArray());
+        return os;
+    }
+
+    @Override
+    public Certificate restoreCertificate() {
+        return certToRestore;
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java]: Can we move the restoreCertificate in bean. It does not have any functionality .. you are just keeping a copy before updating the certificate.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/viewCertDialog.xhtml [hunk 7]
```diff
@@ -123,7 +124,17 @@
                      <div class="form-group col-sm-12">
                         <h:outputText> Note: "-" denotes empty value </h:outputText>
                      </div>
+                     
+                       <div class="form-group">
+                        <div class="col-sm-12">
+                            <div class="form-button-row">
+                                <p:commandButton  id="update" value="UPDATE CERTIFICATE" actionListener="#{certificateBean.updateSelectedCertificate}" update=":certTabView,:certTabView:trustStoreForm,:certTabView:keyStoreForm, viewCertDlgForm:viewCertErrorMsg" styleClass="pull-right"/>
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/viewCertDialog.xhtml]: Do we have all caps button title on the page?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/viewCertDialog.xhtml (hunk 7 above)
Comment (pvenkatakrishnan1) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/viewCertDialog.xhtml]: Done

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java (hunk 2 above)
Comment (pvenkatakrishnan1) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: The methods updateSelectedCertificateKS and updateSelectedCertificate use different service calls to acheive the functionality. These two methods can't be refactored.

**Review by TabassumJafri — APPROVED** [review]

**Review by mhpnguyen — APPROVED** [review]

**Pull request merged**